src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2276,9 +2309,38 @@ def encryption_summary(self, job, id, options):
             results.append({
                 'name': ds_name, 'key_format': ZFSKeyFormat(data['key_format']['value']).value,
                 'key_present_in_database': bool(data['encryption_key']),
-                'valid_key': bool(status['result']), 'locked': data['locked']
+                'valid_key': bool(status['result']), 'locked': data['locked'],
+                'unlock_error': None,
+                'unlock_successful': False,
             })
 
+        failed = set()
+        for ds in sorted(results, key=lambda d: d['name'].count('/')):
+            for i in range(1, ds['name'].count('/') + 1):
+                check = ds['name'].rsplit('/', i)[0]
+                if check in failed:
+                    failed.add(ds['name'])
+                    ds['unlock_error'] = f'Child cannot be unlocked when parent "{check}" is locked'
+
+            if ds['valid_key']:
+                if not ds['unlock_error']:
+                    ds['unlock_successful'] = not bool(ds['unlock_error'])
```

Comment:
Can't it be anything but `True` here?

Reply:
It will be `False` if `unlock_error` is set above in the loop

Reply:
I mean, `not bool(ds['unlock_error'])` will always be `True` in this `if not ds['unlock_error']:` branch?